xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -218,6 +232,35 @@ gd_has_local_address(glusterd_conf_t *priv, const char *hostname)
     return _gf_false;
 }
 
+static int
+glusterd_remote_hostname_new(xlator_t *this, const char *hostname,
```

Comment:
This function is identical to `glusterd_hostname_new()`, and `glusterd_remote_hostname_t` and `glusterd_hostname_t` are identical. Why do we need to duplicate code. I think we can reuse the existing struct and function.